test/app.test.ts

```diff
@@ -163,6 +163,45 @@ test('app with chart dependencies via custom constructs', () => {
 
 });
 
+test('app with nested charts will deduplicate api objects', () => {
+
+  class CustomConstruct extends Construct {
+
+    public obj: ApiObject;
+
+    constructor(scope: Construct, id: string) {
+      super(scope, id);
+
+      this.obj = new ApiObject(this, `${id}obj`, { apiVersion: 'v1', kind: 'CustomConstruct' });
+    }
+  }
+
+  const app = Testing.app();
+  const chart = new Chart(app, 'chart1');
+  const childChart = new Chart(chart, 'chart2');
+  new CustomConstruct(chart, 'child1');
+  new CustomConstruct(childChart, 'child2');
+
+  app.synth();
+
+  expect(fs.readdirSync(app.outdir)).toEqual([
+    '0000-chart1-chart2-c883b207.k8s.yaml',
+    '0001-chart1.k8s.yaml',
+  ]);
+  expect(fs.readFileSync(path.join(app.outdir, '0000-chart1-chart2-c883b207.k8s.yaml'), 'utf8')).toMatchSnapshot();
+  expect(fs.readFileSync(path.join(app.outdir, '0001-chart1.k8s.yaml'), 'utf8')).toMatchSnapshot();
+
+  // individual rendered charts may still share objects
+  expect(Testing.synth(chart)).toMatchObject([
+    { apiVersion: 'v1', kind: 'CustomConstruct', metadata: { name: 'chart1-chart2-child2-child2obj-c828dca6' } },
```

Comment:
Why would this resource be included in the parent chart? Even if its rendered individually, the resource itself isn't being added to this chart, only to the child chart. 

At the end of the day, only two resources are being created here, so I'd expect to only see 2 resources in the final result, here you have 3, 2 of which are duplicated.